bot.js

```diff
@@ -17,47 +19,58 @@ var rules =
 
 // Some words need to be replaced when recomposing the reply so that
 // "I remember my dog" -> "What made you think about *your* dog?"
-var replacements = 
-  [ 
+var replacements =
+  [
     [ "my", "your"],
     [ "My", "your"],
     [ "me", "you" ],
     [ "Me", "you" ],
-    [ "I", "you" ] 
-  ]  
+    [ "I", "you" ]
+  ]
 
-// Checks whether given words match given pattern (pi and wi are
-// indices of current position in pattern and word that we are 
-// looking at). Returns 'null' if matching fails or an array of 
+// Checks whether given words match given pattern (patternIndex and wordIndex are
+// indices of current position in pattern and word that we are
+// looking at). Returns 'null' if matching fails or an array of
 // matched word(s) that correspond to all '*' in the pattern.
-function matchPattern(p, w, pi, wi) {
-  if (pi == p.length && wi == w.length) return [];
-  if (pi == p.length) {
+
+function message(reply) {
```

Comment:
I don't think this function is actually correct - but I suspect this is just something unfinished or leftover from some work?

Reply:
yes exactly! Just leftover work from before. Thank you for your feedback!